frontend/src/views/ProblemPage.tsx

```diff
@@ -26,6 +27,10 @@ function ProblemPage() {
   useEffect(() => {
     getSingleProblem(params.id)
       .then((res) => {
+        res.testCases.forEach((testCase) => {
+          // eslint-disable-next-line no-param-reassign
+          testCase.id = generateRandomId();
+        });
```

Comment:
If the testCases require an id, maybe we could add a quick `private String testCaseId =  UUID.randomUUID().toString();` to the backend and use that instead? That might require an extra line or two in the mapper and mapper test files, but IIRC those kinds of changes don't require an extra test so hopefully it wouldn't be too difficult to implement

Reply:
Ah, the reason I'm not doing that is because we need to generate an ID for the test cases that exist only on the frontend before they are saved as well - this is because a specific ID is needed to make up the Draggable keys, as described [here](https://github.com/atlassian/react-beautiful-dnd/blob/master/docs/api/draggable.md#keys-for-a-list-of-draggable-). We could use a combination of frontend and backend solutions, but that seems superfluous if a frontend solution would work for both - thoughts?

Reply:
Since this is approved I'll merge this now, but we can make changes on a future PR if a backend implementation would still be helpful @alankbi.